src/strategyHelpers/blueGreen/smiBlueGreenHelper.ts

```diff
@@ -233,32 +144,39 @@ export async function validateTrafficSplitsState(
          TRAFFIC_SPLIT_OBJECT,
          getBlueGreenResourceName(name, TRAFFIC_SPLIT_OBJECT_NAME_SUFFIX)
       )
-
+      core.debug(
+         `ts object extracted was ${JSON.stringify(trafficSplitObject)}`
+      )
       if (!trafficSplitObject) {
-         // no traffic split exits
+         core.debug(`no traffic split exits for ${name}`)
          trafficSplitsInRightState = false
+         continue
       }
 
-      trafficSplitObject = JSON.parse(JSON.stringify(trafficSplitObject))
       trafficSplitObject.spec.backends.forEach((element) => {
          // checking if trafficsplit in right state to deploy
          if (element.service === getBlueGreenResourceName(name, GREEN_SUFFIX)) {
-            if (element.weight != MAX_VAL) trafficSplitsInRightState = false
+            trafficSplitsInRightState =
+               trafficSplitsInRightState && element.weight == MAX_VAL
          }
 
          if (
             element.service === getBlueGreenResourceName(name, STABLE_SUFFIX)
          ) {
-            if (element.weight != MIN_VAL) trafficSplitsInRightState = false
+            trafficSplitsInRightState =
+               trafficSplitsInRightState && element.weight == MIN_VAL
          }
       })
    }
-
+   core.debug('returning ' + String(trafficSplitsInRightState))
```

Comment:
this debug message isn't helpful

A user wouldn't understand what "returning " means + there's no context